corehq/apps/domain/forms.py

```diff
@@ -2377,3 +2383,91 @@ def __init__(self, defaults=None, disable_input=False, **kwargs):
+            self.fields['version'].initial = request.GET.get('version')
+        self.helper = FormHelper()
+        self.helper.label_class = 'col-sm-3 col-md-2'
+        self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
+        self.helper.form_tag = False
+
+        self.helper.layout = crispy.Layout(
+            crispy.Field('app_id', id='app-id-search-select'),
+            crispy.Field('location_id', id='location-search-select', css_class="ko-select2"),
+            crispy.Field('version', id='version-input'),
+            hqcrispy.FormActions(
+                crispy.ButtonHolder(
+                    crispy.Button('search', ugettext_lazy("Search"), data_bind="click: search"),
+                    crispy.Button('clear', ugettext_lazy("Clear"), data_bind="click: clear"),
+                    Submit('submit', ugettext_lazy("Add Release Restriction"))
+                )
+            )
+        )
+
+    def app_id_choices(self):
+        choices = ([(None, 'Select Application')])
+        for app in get_brief_apps_in_domain(self.domain):
+            choices.append((app.id, app.name))
+        return choices
+
+    def location_id_choices(self):
+        choices = ([(None, 'Select Location')])
+        for location in SQLLocation.active_objects.filter(domain=self.domain):
+            choices.append((location.location_id, location.name))
+        return choices
+
+    @property
+    def version_build_id(self):
+        app_id = self.cleaned_data['app_id']
+        version = self.cleaned_data['version']
+        return get_version_build_id(self.domain, app_id, version)
+
+    def clean_app_id(self):
+        if not self.cleaned_data.get('app_id'):
+            self.add_error('app_id', _("Please select application"))
+        return self.cleaned_data.get('app_id')
+
+    def clean_location_id(self):
+        if not self.cleaned_data.get('location_id'):
+            self.add_error('location_id', _("Please select location"))
+        return self.cleaned_data.get('location_id')
+
+    def clean_version(self):
+        if not self.cleaned_data.get('version'):
+            self.add_error('version', _("Please enter version"))
+        return self.cleaned_data.get('version')
+
+    def clean(self):
+        app_id = self.cleaned_data.get('app_id')
+        version = self.cleaned_data.get('version')
+        if app_id and version:
+            try:
+                self.version_build_id
+            except BuildNotFoundException as e:
+                self.add_error('version', e)
+
+    def save(self):
+        location_id = self.cleaned_data['location_id']
+        version = self.cleaned_data['version']
+        app_id = self.cleaned_data['app_id']
+        try:
+            LatestEnabledAppRelease.update_status(self.domain, app_id, self.version_build_id, location_id,
+                                                  version, active=True)
+        except ValidationError as e:
+            return False, ','.join(e.messages)
```

Comment:
Nitpick: `','` => `', '` if humans are going to read this.

Reply:
Exceptions don't have a `messages` attribute in Python 3. Best to use `six.text_type(e)` instead.

Reply:
My mistake. It's a ValidationError, so it does have a `messages` attribute.